CAMEL-16307 azure-storage-datalake:...?operation=getFile appends newline

ppalaga opened the pull request:
to the file content

davsclaus commented on core/camel-util/src/main/java/org/apache/camel/util/SkipLastByteInputStream.java:
```diff
@@ -0,0 +1,92 @@
+package org.apache.camel.util;
```
Add license header

ppalaga replied:
Thanks, fixed in 44744b2ad8d83f2e110bd7c789cc504c6b6676ab

davsclaus commented on core/camel-util/src/test/java/org/apache/camel/util/SkipLastNewlineInputStreamTest.java:
```diff
@@ -0,0 +1,56 @@
+package org.apache.camel.util;
```
Add license header

ppalaga replied:
Thanks, fixed in 44744b2ad8d83f2e110bd7c789cc504c6b6676ab

omarsmak reviewed (commented):
@ppalaga a question, I was looking at the datalake client source code and I [saw that you can actually](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-file-datalake_12.3.0/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java#L663) override the default serialization and thus you can override the record separator as explained [here](https://docs.microsoft.com/en-us/java/api/com.azure.storage.file.datalake.datalakefileasyncclient.querywithresponse?view=azure-java-stable). Do you think it would probably make sense to use the other API with the customized query options instead of overriding `BufferedInputStream`?

ppalaga commented:
> @ppalaga a question, I was looking at the datalake client source code and I [saw that you can actually](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-file-datalake_12.3.0/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java#L663) override the default serialization and thus you can override the record separator as explained [here](https://docs.microsoft.com/en-us/java/api/com.azure.storage.file.datalake.datalakefileasyncclient.querywithresponse?view=azure-java-stable). Do you think it would probably make sense to use the other API with the customized query options instead of overriding `BufferedInputStream`?

I thought of that, but I have not found any way in the API docs how it would be possible to actually omit the separator. It only allows changing the separator to some other character which does not seem to help for our use case. But maybe I am missing something and you see a way how we could reach what we need? I'd definitely that option if it exists.

ppalaga commented on core/camel-util/src/main/java/org/apache/camel/util/SkipLastByteInputStream.java:
```diff
@@ -0,0 +1,92 @@
+
+    @Override
+    public synchronized void mark(int readlimit) {
+        throw new UnsupportedOperationException();
```
I think this is safer, because if some subclass overrides our `markSupported()` or callers will ignore our `markSupported()` then they should fail badly instead of being ignored silently.

ppalaga commented on core/camel-util/src/main/java/org/apache/camel/util/SkipLastByteInputStream.java:
```diff
@@ -0,0 +1,92 @@
+
+    @Override
+    public synchronized long skip(long n) throws IOException {
+        throw new UnsupportedOperationException();
```
Yes, you are right, IOException can be removed. UnsupportedOperationException is a runtime exception, so it does not need to be declared.

ppalaga replied:
Fixed in 44744b2ad8d83f2e110bd7c789cc504c6b6676ab

omarsmak commented:
> > @ppalaga a question, I was looking at the datalake client source code and I [saw that you can actually](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-file-datalake_12.3.0/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java#L663) override the default serialization and thus you can override the record separator as explained [here](https://docs.microsoft.com/en-us/java/api/com.azure.storage.file.datalake.datalakefileasyncclient.querywithresponse?view=azure-java-stable). Do you think it would probably make sense to use the other API with the customized query options instead of overriding `BufferedInputStream`?
> 
> I thought of that, but I have not found any way in the API docs how it would be possible to actually omit the separator. It only allows changing the separator to some other character which does not seem to help for our use case. But maybe I am missing something and you see a way how we could reach what we need? I'd definitely be for that option if it exists.

Yeah I know what you mean, the docs are not best. Let me give it a try using your IT and I will report back me myself not sure if it works or not

ppalaga commented:
44744b2:
* Added missing license headers
* Removed superfluous throws declarations

The question is still open whether it is possible to configure the request in such a way that the delimiter is omitted.

omarsmak commented on components/camel-azure-storage-datalake/src/test/java/org/apache/camel/component/azure/storage/datalake/integration/DataLakeProducerIT.java:
```diff
@@ -0,0 +1,113 @@
+import org.apache.camel.component.azure.storage.datalake.DataLakeConstants;
+import org.apache.camel.component.azure.storage.datalake.DataLakeOperationsDefinition;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.assertj.core.api.Assertions;
```
While I was trying the branch, `assertj -core` wasn't added to the `pom.xml`

ppalaga replied:
Good catch, thanks for noticing! I did some experiments with the Azure BOM and I apparently reverted too much when cleaning up.

ppalaga commented:
5719198: 
* Added assertj

omarsmak reviewed (approved):
I thought I will be smarter and override the `recordSeparator` to `/0`. Guess what, at their server side, they seem doing something like `if (recordSeparator == null || recordSeparator == /0) return 
\n;` , then what is the point of specifying the recordSeparator? :D 
Anyway thank you @ppalaga  for this workaround, I will comment on that Azure ticket

ppalaga commented:
Thanks for checking, @omarsmak. We can remove wrapper stream once they add a new endpoint for this.
